presto-cli/src/main/java/com/facebook/presto/cli/JsonPrinter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cli;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.google.common.collect.ImmutableList;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.List;
+
+import static com.facebook.presto.cli.AlignedTablePrinter.formatHexDump;
+import static java.util.Objects.requireNonNull;
+
+public class JsonPrinter
+        implements OutputPrinter
+{
+    private final List<String> fieldNames;
+    private final Writer writer;
+
+    public JsonPrinter(List<String> fieldNames, Writer writer)
+    {
+        this.fieldNames = ImmutableList.copyOf(requireNonNull(fieldNames, "fieldNames is null"));
+        this.writer = requireNonNull(writer, "writer is null");
+    }
+
+    @Override
+    public void printRows(List<List<?>> rows, boolean complete)
+            throws IOException
+    {
+        JsonFactory jsonFactory = new JsonFactory();
+        for (List<?> row : rows) {
+            JsonGenerator jsonGenerator = jsonFactory.createGenerator(writer);
```

Comment:
instead of creating a new JsonGenerator obj for every row,  create a final jsonGenerator above the for loop so it can be reused after each flush stream.